iot_sonoff/views/iot_device_views.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+    <record id="iot_device_form" model="ir.ui.view">
+        <field name="name">iot.device.form</field>
+        <field name="model">iot.device</field>
+        <field name="inherit_id" ref="iot.iot_device_form"/>
+        <field name="arch" type="xml">
+            <header position="inside">
+                <button name="device_run_action" type="object" string="On"
+                        attrs="{'invisible': ['|', ('is_sonoff', '!=', True), ('state', '=', 'sonoff-off')]}"
```

Comment:
I think here 'is_sonoff', '=', False instead of 'is_sonoff', '!=', True is slightly easier to read.